app/forms/steps/income/client/other_work_benefits_form.rb

```diff
@@ -0,0 +1,55 @@
+module Steps
+  module Income
+    module Client
+      class OtherWorkBenefitsForm < Steps::BaseFormObject
+        attribute :applicant_other_work_benefit_received, :value_object, source: YesNoAnswer
+        attribute :amount, :pence
+        attribute :frequency, :value_object, source: PaymentFrequencyType
+
+        validates :applicant_other_work_benefit_received, inclusion: { in: YesNoAnswer.values }
+        validates :amount, numericality: { greater_than: 0 }, if: -> { receives_other_work_benefit? }
+
+        def self.build(crime_application)
+          payment = crime_application.income_payments.work_benefits
+          income = crime_application.income
+          form = new
+
+          form.applicant_other_work_benefit_received = income.applicant_other_work_benefit_received if income
+          form.amount = payment.amount if payment
+
+          form
+        end
+
+        private
+
+        def receives_other_work_benefit?
+          applicant_other_work_benefit_received&.yes?
+        end
+
+        def persist!
+          crime_application.income.update(
+            applicant_other_work_benefit_received:,
+          )
+
+          receives_other_work_benefit? ? persist_income_payment : reset!
+        end
+
+        def persist_income_payment
+          ::IncomePayment.transaction do
+            reset!
+
+            crime_application.income_payments.create!(
+              payment_type: IncomePaymentType::WORK_BENEFITS.value,
+              amount: amount,
+              frequency: PaymentFrequencyType::ANNUALLY,
+            )
+          end
```

Comment:
Yeah potentially, I (think) I've just changed it so that both updates are in the same transaction - but let me know if you think there's a better way of doing it!

Reply:
Following execute for both if and else
- reset!
- update_income_attribute!

What do you think of this?

```suggestion
def persist!
  ::IncomePayment.transaction do
    reset!
    update_income_attribute!

    if receives_other_work_benefit?
      crime_application.income_payments.create!(
        payment_type: IncomePaymentType::WORK_BENEFITS.value,
        amount: amount,
        frequency: PaymentFrequencyType::ANNUALLY,
        )
    end
  end
end
```